### PR TITLE
[RHPAM-4012] upgrade apache-cxf to version 3.4.5

### PR DESCRIPTION
**Pull request by rgdoliveira**
issue: https://issues.redhat.com/browse/RHPAM-4012

related to https://github.com/kiegroup/droolsjbpm-integration/pull/2671

### CODE REVIEW

**Comment by rgdoliveira** [comment]
Jenkins run fdb

**Comment by Ginxo** [comment]
@rgdoliveira duplicated classes at `org.kie:kie-server-spring-boot-autoconfiguration-drools`
```
[2021-12-13T20:42:39.931Z]   Found in:
[2021-12-13T20:42:39.931Z]     javax.validation:validation-api:jar:2.0.1.Final:compile
[2021-12-13T20:42:39.931Z]     jakarta.validation:jakarta.validation-api:jar:2.0.2:compile
```
I will try to reproduce it by executing
`build-chain-action -df 'https://raw.githubusercontent.com/${GROUP:kiegroup}/droolsjbpm-build-bootstrap/${BRANCH:main}/.ci/pull-request-config.yaml' build pr -url https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1848 -sp kiegroup/droolsjbpm-integration`

**Comment by Ginxo** [comment]
@rgdoliveira according to maven dependency tree
```
[INFO] |  |  +- org.apache.cxf:cxf-rt-rs-client:jar:3.4.5:compile
[INFO] |  |  +- jakarta.validation:jakarta.validation-api:jar:2.0.2:compile
[INFO] |  +- org.kie.soup:kie-soup-dataset-api:jar:7.64.0-SNAPSHOT:compile
[INFO] |  |  +- javax.validation:validation-api:jar:2.0.1.Final:compile
```
`jakarta.validation-api:jar:2.0.2` should be excluded from `org.apache.cxf:cxf-rt-rs-client` wdyt?

**Comment by mareknovotny** [comment]
actually i would exclude that `javax.validation:validation-api:jar:2.0.1.Final:compile` pulled from kie-soup

**Comment by rgdoliveira** [comment]
Hi @mareknovotny , I've been talking today with @Ginxo and discussing what we should do in that case.

We were thinking that maybe instead of adding the javax.validation:validation-api exclusion to kie-soup, we could actually move the kie-soup project (kie-soup-dataset-api module) to use jakarta validation-api instead of javax validation-api. What do you think of this approach?
Note that we don't know if more repositories would also require this change.

**Comment by Ginxo** [comment]
> Hi @mareknovotny , I've been talking today with @Ginxo and discussing what we should do in that case.
> 
> We were thinking that maybe instead of adding the javax.validation:validation-api exclusion to kie-soup, we could actually move the kie-soup project (kie-soup-dataset-api module) to use jakarta validation-api instead of javax validation-api. What do you think of this approach? Note that we don't know if more repositories would also require this change.

I would additionally remove javax.validation-api completely from the whole RHBA, wdyt?

**Comment by mareknovotny** [comment]
> > Hi @mareknovotny , I've been talking today with @Ginxo and discussing what we should do in that case.
> > We were thinking that maybe instead of adding the javax.validation:validation-api exclusion to kie-soup, we could actually move the kie-soup project (kie-soup-dataset-api module) to use jakarta validation-api instead of javax validation-api. What do you think of this approach? Note that we don't know if more repositories would also require this change.
> 
> I would additionally remove javax.validation-api completely from the whole RHBA, wdyt?

i think that is not possible due GWT/Errai stack we use and their validation relies on very old javax.validation-api.

So better to try to limit usage everywhere where it is not needed.

**Comment by Ginxo** [comment]
> > > Hi @mareknovotny , I've been talking today with @Ginxo and discussing what we should do in that case.
> > > We were thinking that maybe instead of adding the javax.validation:validation-api exclusion to kie-soup, we could actually move the kie-soup project (kie-soup-dataset-api module) to use jakarta validation-api instead of javax validation-api. What do you think of this approach? Note that we don't know if more repositories would also require this change.
> > 
> > 
> > I would additionally remove javax.validation-api completely from the whole RHBA, wdyt?
> 
> i think that is not possible due GWT/Errai stack we use and their validation relies on very old javax.validation-api.
> 
> So better to try to limit usage everywhere where it is not needed.

so kie-soup for now according to @mareknovotny comment, we should additionally investigate the project/modules affected by this cxf upgrading using kie-soup or validation-api, don't we?

**Comment by mareknovotny** [comment]
i would say only with relation to cxf-core upgrade and if that enforcer complains. Otherwise not related declarations should be changed in another PR change or keep it as it is. We can't change all around the world ;) @Ginxo @rgdoliveira

**Comment by rgdoliveira** [comment]
Jenkins run fdb

**Comment by rgdoliveira** [comment]
Jenkins run fdb

File: pom.xml [hunk 1]
```diff
@@ -3781,6 +3781,10 @@
             <groupId>javax.annotation</groupId>
             <artifactId>javax.annotation-api</artifactId>
           </exclusion>
+         <exclusion>
```
Comment (Ginxo) [inline pom.xml:3784]: be aware of the indentation

Comment (rgdoliveira) [reply pom.xml]: done

**Review by mareknovotny — APPROVED** [review]

**Comment by Ginxo** [comment]
Jenkins run fdb

**Comment by Ginxo** [comment]
Jenkins retest this

**Comment by Ginxo** [comment]
Jenkins execute product fdb

**Comment by Ginxo** [comment]
we should check there's not any remainder cxf:3.3.9 on the prod artifacts (after `FDB prod` job is finished)

**Comment by Ginxo** [comment]
FDB is over and green (yellow but those are random test failures). Let's wait for the productization one.
off-topic: I love this execution summary report :heart: 
```
08:25:35  ::group::[Full Downstream Flow] Execution Summary...
08:25:35  [INFO]  ----------------------------------------------
08:25:35  [INFO]  [kiegroup/droolsjbpm-build-bootstrap]. Execution Result: ok. Time: 21.9s (21946 ms)
08:25:35  [INFO]  [kiegroup/kie-soup]. Execution Result: ok. Time: 3m 19.1s (199134 ms)
08:25:35  [INFO]  [kiegroup/droolsjbpm-knowledge]. Execution Result: ok. Time: 1m 6.8s (66821 ms)
08:25:35  [INFO]  [kiegroup/drools]. Execution Result: ok. Time: 51m 39.5s (3099590 ms)
08:25:35  [INFO]  [kiegroup/optaplanner]. Execution Result: ok. Time: 11m 56.4s (716432 ms)
08:25:35  [INFO]  [kiegroup/lienzo-core]. Execution Result: ok. Time: 12.2s (12221 ms)
08:25:35  [INFO]  [kiegroup/lienzo-tests]. Execution Result: ok. Time: 41.9s (41958 ms)
08:25:35  [INFO]  [kiegroup/appformer]. Execution Result: ok. Time: 40m 40s (2440023 ms)
08:25:35  [INFO]  [kiegroup/kie-uberfire-extensions]. Execution Result: ok. Time: 16.1s (16148 ms)
08:25:35  [INFO]  [kiegroup/jbpm]. Execution Result: ok. Time: 1h 31m 53.9s (5513969 ms)
08:25:35  [INFO]  [kiegroup/droolsjbpm-integration]. Execution Result: ok. Time: 2h 42m 7s (9727078 ms)
08:25:35  [INFO]  [kiegroup/kie-wb-playground]. Execution Result: ok. Time: 1m 18.8s (78899 ms)
08:25:35  [INFO]  [kiegroup/kie-wb-common]. Execution Result: ok. Time: 1h 3m 57.2s (3837253 ms)
08:25:35  [INFO]  [kiegroup/drools-wb]. Execution Result: ok. Time: 22m 8.8s (1328874 ms)
08:25:35  [INFO]  [kiegroup/jbpm-designer]. Execution Result: ok. Time: 1m 56.5s (116578 ms)
08:25:35  [INFO]  [kiegroup/jbpm-work-items]. Execution Result: ok. Time: 18m 46.4s (1126426 ms)
08:25:35  [INFO]  [kiegroup/jbpm-wb]. Execution Result: ok. Time: 10m 36.8s (636881 ms)
08:25:35  [INFO]  [kiegroup/optaplanner-wb]. Execution Result: ok. Time: 2m 38.9s (158972 ms)
08:25:35  [INFO]  [kiegroup/kie-wb-distributions]. Execution Result: ok. Time: 57m 23.7s (3443748 ms)
08:25:35  [INFO]  [kiegroup/openshift-drools-hacep]. Execution Result: ok. Time: 46.6s (46679 ms)
08:25:35  [INFO]  [kiegroup/process-migration-service]. Execution Result: skipped. Time: not defined
08:25:35  [INFO]  [kiegroup/optaweb-employee-rostering]. Execution Result: ok. Time: 10m 46.6s (646651 ms)
08:25:35  [INFO]  [kiegroup/optaweb-vehicle-routing]. Execution Result: ok. Time: 3m 43.9s (223930 ms)
08:25:35  [INFO]  [kiegroup/kie-jpmml-integration]. Execution Result: ok. Time: 56.2s (56206 ms)
08:25:35  [INFO]  ----------------------------------------------
```

**Comment by Ginxo** [comment]
generated artifacts from FDPB contain just cxf:3.4.5 libraries. 
well done @rgdoliveira !!! congrats!

**Pull request merged**